oqamaint/openqabot.py

```diff
@@ -0,0 +1,588 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+
+from collections import namedtuple
+from datetime import date
+import md5
+from pprint import pformat
+import re
+
+
+import requests
+import osc.core
+
+import ReviewBot
+from osclib.comments import CommentAPI
+
+from suse import SUSEUpdate
+
+try:
+    from xml.etree import cElementTree as ET
+except ImportError:
+    from xml.etree import ElementTree as ET
+
+
+try:
+    import simplejson as json
+except ImportError:
+    import json
+
+QA_UNKNOWN = 0
+QA_INPROGRESS = 1
+QA_FAILED = 2
+QA_PASSED = 3
+
+Package = namedtuple('Package', ('name', 'version', 'release'))
+pkgname_re = re.compile(r'(?P<name>.+)-(?P<version>[^-]+)-(?P<release>[^-]+)\.(?P<arch>[^.]+)\.rpm')
+comment_marker_re = re.compile(
+    r'<!-- openqa state=(?P<state>done|seen)(?: result=(?P<result>accepted|declined|none))?(?: revision=(?P<revision>\d+))? -->')
+
+
+class OpenQABot(ReviewBot.ReviewBot):
+
+    """ check ABI of library packages
+    """
+
+    def __init__(self, target_repo, project_settings, api_map, *args, **kwargs):
+
+        self.tgt_repo = target_repo
+        self.project_settings = project_settings
+        self.api_map = api_map
+        super(OpenQABot, self).__init__(*args, **kwargs)
+
+        self.force = False
+        self.openqa = None
+        self.commentapi = CommentAPI(self.apiurl)
+        self.update_test_builds = {}
+        self.pending_target_repos = set()
+        self.openqa_jobs = {}
```

Comment:
Per comments on CLI portion this constructor should be refactored.

Reply:
btw. usage of `cmdln` is the worst thing in our tooling

Reply:
I agree, hence the new code I've added does not use it.

Reply:
Even worse `osc` ships with a copy of `cmdln` nested inside itself. The `osc` tied parts would like be more cumbersome to implement without it given they are built atop `osc`, but anything outside of that surely can.

Reply:
old copy ( and upstream is dead 3 years )